agent_monitor/src/main.cpp

```diff
@@ -415,11 +414,27 @@ std::map<std::string, double> computeIsLookingToward(std::map<std::string, TRBuf
         double angle;
 
         //Get the monitored agent head entity
-        if(agentMonitored=="pr2")
+        //TODO add a rosparam for robot's head joint name
+        if (agentMonitored == "pr2") 
         {
-            monitoredAgentHead = ((Agent*) mapEnts[agentMonitored].back())->skeleton_["head_tilt_link"];
+            std::map<std::string, Joint*> skelMap = ((Agent*) mapEnts[agentMonitored].back())->skeleton_;
+            if (skelMap.find("head_tilt_link") != skelMap.end())
+            {
+              monitoredAgentHead = skelMap["head_tilt_link"];
+            } else {
+              return returnMap;
+            }
+        } else if (agentMonitored == "HERAKLES_HUMAN1" || agentMonitored == "HERAKLES_HUMAN2")
+        {
+            std::map<std::string, Joint*> skelMap = ((Agent*) mapEnts[agentMonitored].back())->skeleton_;
+            if (skelMap.find("head") != skelMap.end())
+            {
+              monitoredAgentHead = skelMap["head"];
+            } else {
+              return returnMap;
+            }
         } else {
-            monitoredAgentHead = ((Agent*) mapEnts[agentMonitored].back())->skeleton_["head"];
```

Comment:
it make no sense to search a "head" if we don't know his joint name.. the best way is to add lots of params for each agent on ros.. that will we be fine for toaster too